libr/io/submap.c

```diff
@@ -0,0 +1,289 @@
+#include <r_io.h>
+#include <r_types.h>
+#include <r_util.h>
+
+ut64 _map_from(RIOMap *map) {
+	return map->itv.addr;
+}
+
+ut64 _map_to(RIOMap *map) {
+	return map->itv.addr + map->itv.size - 1;
+}
+
+//in - incoming
+static bool _submap_find_cb(void *vin, void *vincoming, void *user, int *cmp_res) {
+	RIOSubMap *sm = (RIOSubMap *)vin;
+	ut64 *off = (ut64 *)vincoming;
+
+	if (off[0] < sm->from) {
+		cmp_res[0] = 1;
+		goto beach;
+	}
+	if (off[0] > sm->to) {
+		cmp_res[0] = -1;
+		goto beach;
+	}
+	cmp_res[0] = 0;
+beach:
+	return true;
+}
+
+bool _submap_free_cb(void *user, void *data, ut32 id) {
+	free(data);
+	return true;
+}
+
+//only use this, when no overlapping is guaranteed with the incoming data
+bool _submap_insert_no_overlap_cb(void *vin, void *vincoming, void *user, int *cmp_res) {
```

Comment:
No need for the wrapper

Reply:
it is needed for the insert function. The wrapper allows me to reuse the find_fcn